include/flatbuffers/code_generators.h

```diff
@@ -29,7 +29,7 @@ class BaseGenerator {
     EnsureDirExists(path.c_str());
     if (parser.opts.one_file) return path;
     std::string namespace_dir = path;  // Either empty or ends in separator.
-    auto &namespaces = ns.components;
+    auto namespaces = PrefixedNamespaceComponents(ns, parser.opts.lang);
```

Comment:
ns.components and the value returned by PrefixedNamespaceComponents() are equals for all language other than Java.
To add support for namespace prefix in other languages, I see the following tasks for each language:
- Add new prefix to known_attributes_
- Update Language enum if the language is missing
- Update PrefixedNamespaceComponents() function to retrieve the namespace prefix attribute corresponding to the language being generated
- Replace use of Namespace::component in those generators by PrefixedNamespaceComponents()